packages/api/src/transifexApi.js

```diff
@@ -1,4 +1,33 @@
-import { JsonApi, Resource as JsonApiResource } from '@transifex/jsonapi'; /* eslint-disable-line max-classes-per-file */
+/* eslint-disable no-await-in-loop, max-classes-per-file */
+import { JsonApi, Resource as JsonApiResource } from '@transifex/jsonapi';
+
+function sleep(sec) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, sec * 1000);
+  });
+}
+
+async function download({ interval = 5, ...data }) {
+  // 'this' is a Resource subclass
+  const resource = await this.create({ content_encoding: 'text', ...data });
+  while (!resource.redirect) {
+    /* eslint-disable no-await-in-loop */
+    await sleep(interval);
+    await resource.reload();
+  }
+  return resource.follow();
+}
+
+async function upload({ interval = 5, ...data }) {
+  const resource = await this.create({ content_encoding: 'text', ...data });
```

Comment:
We need to decide if we will be passing the `content_encoding` defaults here or in the calling method. 
Maybe setting the defaults there is better to keep it more abstract, like you do in other places of the code:
`upload.call(this, { content_encoding: 'text', ...data })`